### PR TITLE
chore: deprecate `v1`

### PR DESCRIPTION
**Pull request by duncanista**
### What does this PR do?

Removes `v1` code and its references

### Motivation

"_As of June 1, 2023, AWS CDK v1 has officially reached its end-of-support. Consequently, AWS will no longer provide maintenance, updates, patches, or technical support for this version of the CDK_" – [AWS](https://aws.amazon.com/blogs/devops/cdk-v1-end-of-support/)

### Testing Guidelines

- [x] linting, unit test, and integration tests passing as expected
- [ ] manual test works as expected

### Additional Notes

Release will be kept as `v2`

### Types of Changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [x] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [ ] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -9,10 +9,9 @@ _Please refer to the README.md for information about the structure of this repo
     git clone git@github.com:<your-username>/datadog-cdk-constructs.git
     git checkout -b <my-branch>
     ```
-1. Install dependencies with `yarn install`. Do this within the repo root, as well as within the project version you plan to edit (`v1` or `v2`).
+1. Install dependencies with `yarn install`.
 1. Make your changes.
-1. Run unit tests with `yarn test` (must also be run in `v1` or `v2` root):
-    - If common files were changed, please run unit tests in both `v1` and `v2` folders.
+1. Run unit tests with `yarn test`:
```
Comment (buraizu) [inline CONTRIBUTING.md:14]:
```suggestion
1. Run unit tests with `yarn test`.
```

**Comment by duncanpharvey** [comment]
Should these references to v1 be removed as well?

https://github.com/DataDog/datadog-cdk-constructs/blob/4bd6cae19ce86571e97cd704a90de9abdb11ed58/README.md?plain=1#L21

https://github.com/DataDog/datadog-cdk-constructs/blob/4bd6cae19ce86571e97cd704a90de9abdb11ed58/README.md?plain=1#L36-L41

https://github.com/DataDog/datadog-cdk-constructs/blob/4bd6cae19ce86571e97cd704a90de9abdb11ed58/README.md?plain=1#L50-L53

https://github.com/DataDog/datadog-cdk-constructs/blob/4bd6cae19ce86571e97cd704a90de9abdb11ed58/README.md?plain=1#L64

File: LICENSE [hunk 2]
```diff
@@ -186,7 +187,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright 2021 Datadog
+   Copyright [yyyy] [name of copyright owner]
```
Comment (duncanpharvey) [inline LICENSE:190]: Should be 2024 Datadog?

**Review by duncanpharvey — APPROVED** [review]
LGTM with some minor comments

File: LICENSE (hunk 2 above)
Comment (duncanista) [reply LICENSE:190]: I think that was when this was created!

Comment (duncanista) [reply LICENSE:190]: NVM – im tweaking, fixing this

**Pull request merged**